packages/react-day-picker/src/components/Table/Table.test.tsx

```diff
@@ -0,0 +1,43 @@
+import * as React from 'react';
+
+import { enGB, enUS } from 'date-fns/locale';
+import { customRender, PageObjects } from 'test';
+
+import { Table } from './Table';
+
+const po = new PageObjects(new Date());
+
+describe('It should render weeks at the boundaries of years correctly', () => {
+  test('If the first days of the month start at week 52 from the previous year, week 52 should be rendered on the 1st row', () => {
+    customRender(<Table displayMonth={new Date(2022, 0)} />, { locale: enGB });
+    const week1 = po.weeks[0].querySelectorAll('td');
+    const day6 = week1[5];
+    const day7 = week1[6];
+    expect(day6).toHaveTextContent('11st January (Saturday)');
+    expect(day7).toHaveTextContent('22nd January (Sunday)');
+  });
+
+  test('If there is a week 53, and we show week 1 afterwards, week 1 should be rendered on the 6th row', () => {
+    customRender(<Table displayMonth={new Date(2022, 11)} />, {
+      locale: enUS,
+      showOutsideDays: true,
+      fixedWeeks: true
+    });
+    const week6 = po.weeks[5].querySelectorAll('td');
+    const day36 = week6[0];
+    const day37 = week6[1];
+    expect(day36).toHaveTextContent('11st January (Sunday)');
+    expect(day37).toHaveTextContent('22nd January (Monday)');
+  });
+
+  test('If the last week of the year is 1, it should be rendered on the 5th row', () => {
+    customRender(<Table displayMonth={new Date(2021, 11)} />, {
+      locale: enUS
+    });
+    const week5 = po.weeks[4].querySelectorAll('td');
+    const day29 = week5[0];
+    const day30 = week5[1];
+    expect(day29).toHaveTextContent('2626th December (Sunday)');
+    expect(day30).toHaveTextContent('2727th December (Monday)');
```

Comment:
This test passes with the previous code because of this code:
```
if (week === 1 && getMonth(date) === 11) {
  week = 53;
}
```
With the proposed code changes, the above code has been removed, but the test still passes, and the other two edge case tests do too.

However, removing that code above now renders the last month of the year with a week number of one (see screenshot):
<img width="1125" alt="Screen Shot 2021-11-02 at 3 09 22 PM" src="https://user-images.githubusercontent.com/2838338/139929462-ec09bfa5-c124-41fc-a561-ca9ba6303383.png">

I am making the assumption that the reason for the code above was to fix the original ordering bug, not to apply any desired transformations to the week numbers.

The week number, 1, without the transformation to map it to 53, is now consistent between the December 2021 and the January 2022 calendars, so I think it's a reasonable change of behavior to the rendering of the week numbers that also fixes these other edge cases. Additionally, making week number transformations significantly increases the complexity of the definition of a week number, so removing the transformation is an improvement as far as reducing complexity is concerned.

Reply:
As long as it's documented in the (eventual) library docs, I don't see that as a breaking change. It's a lot easier to understand "week # = latest week # within this 7 day period" than the edge case of the end of the year.

The only thing that gives me pause is that a week 53 is extremely rare but technically possible. However, the number 53 isn't special in itself. Are there any known use cases where someone has to detect whether something is the last week, and is currently reliant on looking for a hardcoded value of 52/53 ?

Reply:
@tjfred35 are you sure from the screen above the week numbers are correct? For December 2021 , my calendar app shows 48–1 for (week starting from Monday) or 47–52 (week starting from Sunday). But never the first week is 49. We should double check this...

Reply:
Week numbers are wild, I guess! On the published beta in codesandbox I can get the following

**locale={enUS}**
<img width="352" alt="Screen Shot 2021-11-04 at 8 34 40 AM" src="https://user-images.githubusercontent.com/2838338/140314540-d83ffcca-dddb-43eb-884a-d60baddcfb3a.png">

**locale={enGB}**
<img width="347" alt="Screen Shot 2021-11-04 at 8 37 53 AM" src="https://user-images.githubusercontent.com/2838338/140314661-1ee249c7-7311-46c6-9fd0-e67ebded84e0.png">

https://codesandbox.io/s/daypicker-basics-customization-weeknumber-example-forked-k9o5n

What locale / configuration do you use to get 47-52?

Reply:
> What locale / configuration do you use to get 47-52?

Interesting. The "Region" in my Mac OS is set to Spain. Indeed the week number changes. If there are issues we will find them out but it looks like it is expected :)